File: Models/Entities/Event.cs
using SEAL.NET.Models.Enums;

namespace SEAL.NET.Models.Entities
{
    public class Event
    {
        public Guid EventId { get; set; } = Guid.NewGuid();
        public string EventName { get; set; } = string.Empty;
        public string? Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public EventStatus Status { get; set; } = EventStatus.Upcoming;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public bool IsPublished { get; set; } = false;
        public bool IsArchived { get; set; } = false;
        public DateTime? RegistrationClosedAt { get; set; }
        public DateTime? JudgingStartedAt { get; set; }
        public DateTime? JudgingEndedAt { get; set; }


        public List<Category> Categories { get; set; } = new();
        public List<Round> Rounds { get; set; } = new();
        public List<EventRegistration> Registrations { get; set; } = new();
    }
}

File: Frontend/types/event.ts
export type EventStatus =
  | "Draft"
  | "Upcoming"
  | "RegistrationClosed"
  | "Judging"
  | "RankingPublished"
  | "Ongoing"
  | "Completed"
  | "Cancelled"
  | "Archived";

export type RoundStatus = "Draft" | "Open" | "Closed" | "Locked" | "ResultsPublished";

export interface Event {
  eventId: string;
  eventName: string;
  description?: string;
  status: EventStatus;
  startDate: string;
  endDate: string;
  isPublished?: boolean;
  isArchived?: boolean;
  registrationClosedAt?: string | null;
  judgingStartedAt?: string | null;
  judgingEndedAt?: string | null;
  categories?: Category[];
  rounds?: Round[];
}

export interface Category {
  categoryId: string;
  categoryName: string;
  description?: string;
  eventId?: string;
  teamCount?: number;
}

export interface Round {
  roundId: string;
  roundName: string;
  submissionDeadline?: string | null;
  roundOrder: number;
  maxTeamsAdvancing: number;
  eventId?: string;
  status?: RoundStatus;
  isRankingPublished?: boolean;
  isSubmissionLocked?: boolean;
}

export interface Criteria {
  criteriaId: string;
  criteriaName: string;
  maxScore: number;
  weight: number;
  roundId: string;
}

export interface EventPayload {
  eventName: string;
  description?: string;
  startDate: string;
  endDate: string;
  status: EventStatus;
}

export interface CategoryPayload {
  categoryName: string;
  description?: string;
}

export interface RoundPayload {
  roundName: string;
  submissionDeadline: string;
  roundOrder: number;
  maxTeamsAdvancing: number;
}

export interface CriteriaPayload {
  criteriaName: string;
  maxScore: number;
  weight: number;
}

export interface ApiResponse<T = void> {
  message: string;
  data?: T;
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  pageSize: number;
}
